### ISSUE TITLE
Gutenberg_Block_Template 'theme' property is confusing, because templates might be added by plugins instead of themes

### ISSUE THREAD

**Issue by Aljullu** [WordPress/gutenberg#37401]
### Description

Currently, the `Gutenberg_Block_Template` has a [`theme` property](https://github.com/WordPress/gutenberg/blob/8eb1f341285362f6a131fcba81a8d054b6b5b5c5/lib/compat/wordpress-5.9/class-gutenberg-block-template.php#L21-L26) that is used to indicate which theme the template belongs to. That works great for themes, but it's confusing when plugins add templates: should plugin devs set the plugin name in the `theme` property? Currently, the answer is yes, but it's confusing!

I wonder if `theme` property could be renamed to something that encompasses themes and plugins. So a template added by a plugin, instead of being defined like this:

```PHP
...
$template->origin      = 'plugin';
$template->theme       = 'woocommerce//woocommerce';
...
```

would be defined like something along these lines:
```PHP
...
$template->origin      = 'plugin';
$template->origin_slug = 'woocommerce//woocommerce';
...
```

I know we are late in the WP 5.9 release cycle so we might not be able to make a change like that at this point. In that case, I would like to clarify if the recommended way for plugins to add templates is to use the `theme` property.

Any thoughts on this @carolinan @talldan @kevin940726?

### Step-by-step reproduction instructions

-

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
Thanks for creating the issue. I've been meaning to create one along the same lines.

It's probably a bit too late to change this in the release cycle for WordPress 5.9, and the REST API for templates has been available since WordPress 5.8, so there will be backward compatibility concerns with any change.

I think what we do need is some documentation outlining exactly how plugins can add templates before the release of 5.9.

At the moment I think the agreed settings are for `source` to be set to 'plugin' and `theme` to be set to the slug of the plugin. `origin` is derived from `source` when a template is customized.

**Comment by Aljullu** [comment]
Makes sense, thanks for clarifying this part, @talldan:

> At the moment I think the agreed settings are for `source` to be set to 'plugin' and `theme` to be set to the slug of the plugin. `origin` is derived from `source` when a template is customized.

Based on that I will close this issue assuming it works as expected. Looking forward to the documentation. :slightly_smiling_face:
